lib/sqlalchemy/dialects/oracle/base.py

```diff
@@ -1290,7 +1290,7 @@ def initialize(self, connection):
         super(OracleDialect, self).initialize(connection)
 
         self.implicit_returning = self.__dict__.get(
-            "implicit_returning", self.server_version_info > (10,)
+            "implicit_returning", self.server_version_info and self.server_version_info > (10,)
```

Comment:
server_version_info should never be None here since there is a cx_Oracle database connection.